mowgli_etl/pipeline/wdc/wdc_size_buckets.py

```diff
@@ -0,0 +1,149 @@
+                new_object[a] = {"value": None, "unit": None}
+                new_object[a]["value"] = (
+                    getattr(generic.dimension, a).value
+                    + getattr(local.dimension, a).value * local.count
+                ) / (local.count + 1)
+                new_object[a]["unit"] = getattr(generic.dimension, a).unit
+
+        updated_generic = {
+            "name": generic.name,
+            "dimension": new_object,
+            "count": local.count + 1,
+            "unit": generic.unit,
+        }
+
+        return WdcGenericSize.from_dict(updated_generic)
+
+    def generalize(self, wdc_product: Tuple[WdcProductType, WdcProductDimensions]):
+        # General units: in^3, V, lb
+        product_type = wdc_product[0]
+        product_dimensions = wdc_product[1]
+        dimension = None
+        unit = None
+        if (
+            product_dimensions.depth
+            or product_dimensions.height
+            or product_dimensions.length
+            or product_dimensions.width
+        ):
+            dimension = dict()
+            if product_dimensions.depth:
+                dimension["depth"] = {
+                    "value": self.__convert(
+                        product_dimensions.depth.value, product_dimensions.depth.unit
+                    ),
+                    "unit": "in",
+                }
+            if product_dimensions.height:
+                dimension["height"] = {
+                    "value": self.__convert(
+                        product_dimensions.height.value, product_dimensions.height.unit
+                    ),
+                    "unit": "in",
+                }
+            if product_dimensions.length:
+                dimension["length"] = {
+                    "value": self.__convert(
+                        product_dimensions.length.value, product_dimensions.height.unit
+                    ),
+                    "unit": "in",
+                }
+            if product_dimensions.width:
+                dimension["width"] = {
+                    "value": self.__convert(
+                        product_dimensions.width.value, product_dimensions.width.unit
+                    ),
+                    "unit": "in",
+                }
+            unit = "in^3"
+        elif product_dimensions.power:
+            dimension = {
+                "value": self.__convert(
+                    product_dimensions.power.value, product_dimensions.power.unit
+                ),
+                "unit": "V",
+            }
+            unit = "v'"
+        elif product_dimensions.weight:
+            dimension = {
+                "value": self.__convert(
+                    product_dimensions.weight.value, product_dimensions.weight.unit
+                ),
+                "unit": "lb",
+            }
+            unit = "lb"
+
+        if not dimension:
+            return None
+        generic_dimensions = self.__update(
+            WdcGenericSize(
```

Comment:
I see what this is for now. How about `WdcProductSize` or `WdcProductEstimatedSize`.